interface/product_registration/product_registration_service.php

```diff
@@ -0,0 +1,100 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>;.
+ *
+ * @package OpenEMR
+ * @author  Matthew Vita <matthewvita48@gmail.com>
+ * @link    http://www.open-emr.org
+ */
+
+$fake_register_globals=false;
+$sanitize_all_escapes=true;
+include_once("../globals.php");
+require_once($GLOBALS['fileroot'] . "/interface/main/exceptions/invalid_email_exception.php");
+require_once($GLOBALS['fileroot'] . "/interface/product_registration/local_product_registration_exception.php");
+require_once($GLOBALS['fileroot'] . "/interface/product_registration/remote_product_registration_exception.php");
+require_once($GLOBALS['fileroot'] . "/ccr/uuid.php");
+
+// TODO: @Brady to help me get this in the migration DSL...
+//CREATE TABLE product_registration (
+//    registration_id CHAR(36) NULL,
+//    email VARCHAR(255) NULL,
+//    opt_out TINYINT(1) NULL,
+//    PRIMARY KEY (registration_id)
+//);
+
+class ProductRegistrationService {
+    private $sqlQuery;
+    private $sqlStatement;
+
+    public function __construct($sqlQuery, $sqlStatement) {
+        $this->sqlQuery = $sqlQuery;
+        $this->sqlStatement = $sqlStatement;
+    }
+
+    public function getProductStatus() {
+        $row = call_user_func($this->sqlQuery, 'SELECT * FROM product_registration LIMIT 1');
+
+        $payload = new stdClass();
+
+        if (empty($row)) {
+            $payload->status = 'UNREGISTERED';
+        } else if (!(empty($row['registration_id']))) {
+            $payload->status = 'REGISTERED';
+            $payload->email = $row['email'];
+            $payload->registration_id = $row['registration_id'];
+        } else if (!(empty($row['opt_out'])) && $row['opt_out'] == true) {
+            $payload->status = 'OPT_OUT';
+        }
+
+        return $payload;
+    }
+
+    public function registerProduct($email) {
+        if (!$email || $email == 'false') {
+            $this->optOutStrategy();
+            return null;
+        } else {
+            return $this->optInStrategy($email);
+        }
+    }
+
+    private function optInStrategy($email) {
+        if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
+            throw new InvalidEmailException($email);
+        }
+
+        $uuid = getUuid();
+
+        $result = call_user_func($this->sqlStatement,
+                                 'INSERT INTO product_registration (registration_id, email, opt_out) VALUES (?, ?, ?)',
+                                 array($uuid, $email, false));
+
+        // TODO: how can I check the outcome of the response?
+        if (empty($result)) {
+            throw new LocalProductRegistrationException("Database issue... can't register product.");
+        }
```

Comment:
Note that this block of code(above 3 lines) will never get called since sqlStatement will die if it errors out